src/base_alloc/base_alloc_windows.c

```diff
@@ -5,8 +5,38 @@
  * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 */
 
+#include "base_alloc.h"
 #include <windows.h>
 
+static void umf_ba_constructor(void) {}
```

Comment:
You can avoid all the changes in this file by just adding:
```
#if defined(_WIN32) && defined(UMF_SHARED_LIBRARY)
atexit(umf_ba_destroy_global);
#endif
```

to `umf_ba_create_global` in base_alloc_global.c but see my comment below about the atexit.